sky/shell/platform/mojo/dart_tracing.cc

```diff
@@ -0,0 +1,144 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sky/shell/platform/mojo/dart_tracing.h"
+
+#include <utility>
+
+#include "dart/runtime/include/dart_tools_api.h"
+#include "mojo/public/cpp/application/application_impl.h"
+
+namespace dart {
+
+void DartTimelineController::Enable(const mojo::String& categories) {
+  if (categories == mojo::String("Dart")) {
+    Dart_GlobalTimelineSetRecordedStreams(DART_TIMELINE_STREAM_DART);
+  } else {
+    // TODO(johnmccutchan): Respect |categories|.
+    EnableAll();
+  }
+}
+
+void DartTimelineController::EnableAll() {
+  Dart_GlobalTimelineSetRecordedStreams(DART_TIMELINE_STREAM_DART);
+  Dart_GlobalTimelineSetRecordedStreams(DART_TIMELINE_STREAM_ALL |
```

Comment:
I believe you can replace both these lines with just `Dart_GlobalTimelineSetRecordedStreams(DART_TIMELINE_STREAM_ALL)`